ocaml/typing/env.ml

```diff
@@ -3298,26 +3378,53 @@ let lookup_all_dot_constructors ~errors ~use ~loc usage l s env =
 (* General forms of the lookup functions *)
 
 let lookup_module_path ~errors ~use ~loc ~load lid env : Path.t =
-  match lid with
-  | Lident s ->
-      if !Clflags.transparent_modules && not load then
-        fst (lookup_ident_module Don't_load ~errors ~use ~loc s env)
-      else
-        fst (lookup_ident_module Load ~errors ~use ~loc s env)
-  | Ldot(l, s) -> fst (lookup_dot_module ~errors ~use ~loc l s env)
-  | Lapply _ as lid ->
-      let path_f, _comp_f, path_arg = lookup_apply ~errors ~use ~loc lid env in
-      Papply(path_f, path_arg)
+  let path, locks =
+    match lid with
+    | Lident s ->
+        if !Clflags.transparent_modules && not load then
+          let path, locks, _ =
+            lookup_ident_module Don't_load ~errors ~use ~loc s env
+          in
+          path, locks
+        else
+          let path, locks, _ =
+            lookup_ident_module Load ~errors ~use ~loc s env
+          in
+          path, locks
+    | Ldot(l, s) ->
+        let path, locks, _ = lookup_dot_module ~errors ~use ~loc l s env in
+        path, locks
+    | Lapply _ as lid ->
+        let path_f, _comp_f, path_arg = lookup_apply ~errors ~use ~loc lid env in
+        Papply(path_f, path_arg), []
+  in
+  if use then begin
+    let mda_mode = Mode.Value.legacy |> Mode.Value.disallow_right in
```

Comment:
I assume this gets replaced in a later patch. Probably better to define it next to the type definition for the type with the `mda_` fields.